src/tests/ftest/util/server_utils.py

```diff
@@ -306,24 +324,40 @@ def prepare_storage(self, user, using_dcpm=None, using_nvme=None):
             ServerFailed: if there was an error preparing the storage
 
         """
+        self._prepare_server_files()
+
         cmd = DaosServerCommand(self.manager.job.command_path)
         cmd.sudo = False
         cmd.debug.value = False
-        cmd.set_sub_command("storage")
-        cmd.sub_command_class.set_sub_command("prepare")
-        cmd.sub_command_class.sub_command_class.target_user.value = user
-        cmd.sub_command_class.sub_command_class.force.value = True
 
         # Use the configuration file settings if no overrides specified
         if using_dcpm is None:
             using_dcpm = self.manager.job.using_dcpm
         if using_nvme is None:
             using_nvme = self.manager.job.using_nvme
 
-        if using_dcpm and not using_nvme:
-            cmd.sub_command_class.sub_command_class.scm_only.value = True
-        elif not using_dcpm and using_nvme:
-            cmd.sub_command_class.sub_command_class.nvme_only.value = True
+        if using_dcpm:
+            # Prepare SCM storage
+            cmd.set_sub_command("scm")
+            cmd.sub_command_class.set_sub_command("prepare")
+            cmd.sub_command_class.sub_command_class.target_user.value = user
+            result = run_remote(
+                self.log, self._hosts, cmd.with_exports, timeout=self.storage_prepare_timeout.value)
+            if not result.passed:
+                # Add some debug due to the failure
+                run_remote(self.log, self._hosts, "sudo -n ipmctl show -v -dimm")
+                run_remote(self.log, self._hosts, "ndctl list")
+                raise ServerFailed("Error preparing dcpm storage")
+        if using_nvme:
+            # Prepare NVMe storage
+            cmd.set_sub_command("nvme")
+            cmd.sub_command_class.set_sub_command("prepare")
+            cmd.sub_command_class.sub_command_class.target_user.value = user
+            result = run_remote(
+                self.log, self._hosts, cmd.with_exports, timeout=self.storage_prepare_timeout.value)
+            if not result.passed or "No SCM modules detected; skipping operation" in \
```

Comment:
minor: I'm not sure if the `or "No SCM modules detected..."` clause is still valid here. I don't think it will hurt anything, but it's probably cruft now. @tanabarr: This message wouldn't be displayed by the new `nvme prepare` command, right?

Reply:
correct that shouldn't be being displayed by `nvme prepare` command